src/fw/kernel/event_loop.c

```diff
@@ -147,11 +147,26 @@ bool launcher_popups_are_blocked(void) {
   return s_block_popup_count > 0;
 }
 
+// FIRM-425: sometimes, if the system goes out to lunch for a long time when
+// loading a watchface as a result of pressing 'back', the
+// FORCE_QUIT_HOLD_MS timer will trigger.  Check once we wake up and process
+// the launcher_force_quit_app event to see if the back button release event
+// got processed between the NewTimer firing and KernelMain waking up to do the kill.
+static bool s_force_quit_was_cancelled = false;
+
 void launcher_cancel_force_quit(void) {
+  s_force_quit_was_cancelled = true;
   new_timer_stop(s_back_hold_timer);
 }
 
 static void launcher_force_quit_app(void *data) {
+  if (s_force_quit_was_cancelled) {
+    // If you see this in logs after FIRM-556 (general system sluggishness)
+    // is fixed, please file a bug!
+    PBL_LOG(LOG_LEVEL_ERROR, "NewTimer event fired for force quit, but the back button was released before we went to deal with it!  Wow, we must have been really slow!  Please file a bug!");
+    return;
```

Comment:
it would probably be good to track this on memfault, so we have an idea on how often this happens

Reply:
Seems right.  Glued that in.